### ISSUE TITLE
[Enhancement Proposal] Support `--subscription` in `az login`

### ISSUE THREAD

**Issue by jiasli** [Azure/azure-cli#14933]
**Is your feature request related to a problem? Please describe.**

Sometimes, the user's account has access to many subscriptions (for example, over 300) (https://github.com/MicrosoftDocs/azure-docs-cli/issues/2148).

**Describe the solution you'd like**

A parameter `--subscription` can be added to explicitly select the subscription without calling [Subscriptions - List](https://docs.microsoft.com/en-us/rest/api/resources/subscriptions/list) API. 

It must be used together with `--tenant` so that CLI doesn't need to find **in which tenant the subscription resides**, as finding that will require calling [Tenants - List](https://docs.microsoft.com/en-us/rest/api/resources/tenants/list) and [Subscriptions - List](https://docs.microsoft.com/en-us/rest/api/resources/subscriptions/list) APIs.

`az login` will look like:

```console
> az login --tenant 54826b22-38d6-4fb2-bad9-b7b93a3e9c5a --subscription 0b1f6471-1bf0-4dda-aec3-cb9272f09590
[
  {
    "cloudName": "AzureCloud",
    "homeTenantId": "54826b22-38d6-4fb2-bad9-b7b93a3e9c5a",
    "id": "0b1f6471-1bf0-4dda-aec3-cb9272f09590",
    "isDefault": false,
    "managedByTenants": [
      {
        "tenantId": "2f4a9838-26b7-47ee-be60-ccc1fdec5953"
      }
    ],
    "name": "AzureSDKTest",
    "state": "Enabled",
    "tenantId": "54826b22-38d6-4fb2-bad9-b7b93a3e9c5a",
    "user": {
      "name": "username@microsoft.com",
      "type": "user"
    }
  }
]
```

**Comment by yonzhan** [comment]
add to S`175

**Comment by jiasli** [comment]
I am more and more inclined to make `--subscription` or `az account set` explicit and mandatory.

Currently, `az login` chooses the first enabled subscription as the default subscription:

https://github.com/Azure/azure-cli/blob/e8efb791ea2f0ba5e4f172b26f75725fd8aa8079/src/azure-cli-core/azure/cli/core/_profile.py#L500-L502

Without explicitly specifying the subscription, the user may accidentally perform operations on a wrong subscription. For example:

1. Running `az group delete` on a wrong subscription can cause production environment destruction.
1. Running `az role assignment` (which by default uses the current subscription as `--scope` (https://github.com/Azure/azure-cli/issues/24753)) can lead to permission leak.

See: IcM 418947258

**Comment by lanni-energinet** [comment]
@jiasli 

Any updates on this issue. We are using Azure/login github action and the login process could definitely be sped up by AZ-CLI skipping list subscriptions?

**Comment by blueboxes** [comment]
I have a number of power shell script people use locally and systems with hundreds of subscriptions. Setting the subscription by name or id would really help

**Comment by tonur** [comment]
Regarding interactive oneshot login with az, a workaround that worked for me is to pipe something into the `az login` command. This of course will not help if it is slow to list subscriptions, but if it just needs to select the default subscription in order to set it manually with `az account select --subscription $x`, then this works wonders for me:
`echo '' | az login`
I can use it in my script without selecting anything with the following:
`echo '' | az login --output none && az account select --subscription $x`

EDIT: Sadly, this has stopped working for me, and requires a fresh `az login` that does not accept --output none or any stdin, otherwise it will just fail. I hope that Microsoft will add the feature of setting the subscription on login in the future.

**Comment by brucetan-displayr** [comment]
any updates on this? It would be really useful and speed up the az login process

**Comment by anton-x-t** [comment]
Any update on this? This would speed up my daily work not having to manually select the subscription every time I log in via CLI.

**Comment by quanterion** [comment]
Yeah, my team wants it as well not because listing subscriptions can be slow but for automation to remove manual step to select a specific subscription in our dev workflow

**Comment by drmcclelland** [comment]
This would be very helpful for us as well - a single `az login --subscription` command would help make things more automated, quicker, and more concise in our scripts.

**Comment by Alex-AZPS** [comment]
https://techcommunity.microsoft.com/blog/azuretoolsblog/faster-az-login-introducing---skip-subscription-discovery-and-targeted---subscri/4526116  This feature has been released; please give it a try. I am closing this issue; if there are any other issues, please open a new ticket.
